### PR TITLE
Fix port in use while use jmx jvm argument

### LINKED ISSUE #15168: Add JMX metrics encountered a bind failed
**Alluxio Version:**

Master branch, commit hash 2f0c99044dfd7f8578bd3cc5606b4c0ef5b8a7a7

**Describe the bug**
```
Error: Exception thrown by the agent : java.rmi.server.ExportException: Port already in use: 1102; nested exception is: 
        java.net.BindException: Address already in use (Bind failed)
sun.management.AgentConfigurationError: java.rmi.server.ExportException: Port already in use: 1102; nested exception is: 
        java.net.BindException: Address already in use (Bind failed)
        at sun.management.jmxremote.ConnectorBootstrap.startRemoteConnectorServer(ConnectorBootstrap.java:480)
        at sun.management.Agent.startAgent(Agent.java:262)
        at sun.management.Agent.startAgent(Agent.java:452)
Caused by: java.rmi.server.ExportException: Port already in use: 1102; nested exception is: 
        java.net.BindException: Address already in use (Bind failed)
        at sun.rmi.transport.tcp.TCPTransport.listen(TCPTransport.java:346)
        at sun.rmi.transport.tcp.TCPTransport.exportObject(TCPTransport.java:254)
        at sun.rmi.transport.tcp.TCPEndpoint.exportObject(TCPEndpoint.java:412)
        at sun.rmi.transport.LiveRef.exportObject(LiveRef.java:147)
        at sun.rmi.server.UnicastServerRef.exportObject(UnicastServerRef.java:237)
        at sun.rmi.registry.RegistryImpl.setup(RegistryImpl.java:213)
        at sun.rmi.registry.RegistryImpl.<init>(RegistryImpl.java:173)
        at sun.management.jmxremote.SingleEntryRegistry.<init>(SingleEntryRegistry.java:49)
        at sun.management.jmxremote.ConnectorBootstrap.exportMBeanServer(ConnectorBootstrap.java:816)
        at sun.management.jmxremote.ConnectorBootstrap.startRemoteConnectorServer(ConnectorBootstrap.java:468)
        ... 2 more
Caused by: java.net.BindException: Address already in use (Bind failed)
        at java.net.PlainSocketImpl.socketBind(Native Method)
        at java.net.AbstractPlainSocketImpl.bind(AbstractPlainSocketImpl.java:387)
        at java.net.ServerSocket.bind(ServerSocket.java:390)
        at java.net.ServerSocket.<init>(ServerSocket.java:252)
        at java.net.ServerSocket.<init>(ServerSocket.java:143)
        at sun.rmi.transport.proxy.RMIDirectSocketFactory.createServerSocket(RMIDirectSocketFactory.java:45)
        at sun.rmi.transport.proxy.RMIMasterSocketFactory.createServerSocket(RMIMasterSocketFactory.java:345)
        at sun.rmi.transport.tcp.TCPEndpoint.newServerSocket(TCPEndpoint.java:670)
        at sun.rmi.transport.tcp.TCPTransport.listen(TCPTransport.java:335)
        ... 11 more
```
**To Reproduce**

Add the following content into `alluxio-env.sh` and execute `bin/alluxio-start.sh master`

```
ALLUXIO_MASTER_JAVA_OPTS="-Dcom.sun.management.jmxremote \
-Dcom.sun.management.jmxremote.port=1102 \
-Dcom.sun.management.jmxremote.local.only=false \
-Dcom.sun.management.jmxremote.authenticate=false \
-Dcom.sun.management.jmxremote.ssl=false"
```

**Expected behavior**

Master should started successfully, but report `Address already in use`

**Urgency**

High

**Are you planning to fix it**

No

**Additional context**
No

### PR DESCRIPTION
**Pull request by maobaolong**
Fix #15168

### CODE REVIEW

File: bin/alluxio-monitor.sh [hunk 1]
```diff
@@ -56,10 +56,10 @@ get_env() {
 
   # Remove the remote debug configuration to avoid the error: "transport error 20: bind failed: Address already in use."
   # See https://github.com/Alluxio/alluxio/issues/10958
```
Comment (Xenorith) [inline bin/alluxio-monitor.sh:58]: add the issue url https://github.com/Alluxio/alluxio/issues/15168 to this comment?

Comment (maobaolong) [reply bin/alluxio-monitor.sh:58]: Done

**Comment by maobaolong** [comment]
@Xenorith Thanks for your review, please take another look, thank you

**Review by Xenorith — APPROVED** [review]

**Comment by Xenorith** [comment]
alluxio-bot, merge this please

**Comment by maobaolong** [comment]
@Xenorith Thanks for merge this PR!

BTW, I just feel curious why add a `^` before the regex for the `-agentlib:jdwp=transport=dt_socket.*address=[0-9]*`, if we added the `^` as it is now, if `-agentlib:jdwp=transport=dt_socket.*address=[0-9]*` appears in the middle of the `ALLUXIO_XXX_JAVA_OPTS`, it cannot be removed i think.

**Comment by Xenorith** [comment]
i think you are right. i dont see a reason why it must be the beginning of the string
when reading more about it, i also find that the regex we use may not be complete and result in incorrect replacement. for example if i add a timeout value like `-agentlib:jdwp=transport=dt_socket,server=y,address=localhost:8000,timeout=5000`, then the sed will leave behind `,timeout=5000`

**Pull request merged**